server/encoder/encoder_settings.cpp

```diff
@@ -141,7 +143,7 @@ static constexpr auto ffmpeg_version()
 	return result;
 }
 
-static std::optional<wivrn::video_codec> filter_codecs_vaapi(wivrn_vk_bundle & bundle, const std::vector<wivrn::video_codec> & codecs)
+static std::optional<wivrn::video_codec> filter_codecs_vaapi(wivrn_vk_bundle & bundle, const std::vector<wivrn::video_codec> & codecs, int bit_depth)
 {
 	video_encoder_ffmpeg::mute_logs mute;
 	encoder_settings s{
```

Comment:
bit_depth is unset in the structure, auto probing will fail. We will hit the throw in the constructor because it's neither 8 or 10.

Reply:
This is the code to detect supported codecs, used when encoders are not specified in the configuration file.

It will try to create vaapi decoders, if an exception is thrown it marks the codec as unsupported.

Reply:
Just add after initialisation
```c++
s.bit_depth = bit_depth;
```
Or you can shuffle around fields so that it can be written in the initialiser list. Because of the base class it is not possible to use designated initialisers here and there's fields in between we want to keep at the default value.